### PR TITLE
querydb: add new kotlin query `nopTrustManagerUsed`

### PR DESCRIPTION
**Pull request by ursachec**

### CODE REVIEW

File: querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        val nopTrustManagerFullNames =
+          cpg.typeDecl
+            .isExternal(false)
+            .filter(_.inheritsFromTypeFullName.contains("javax.net.ssl.X509TrustManager"))
+            .filter { node =>
+              node.method.name("checkClientTrusted").block.expressionDown.isEmpty ||
+              node.method.name("checkServerTrusted").block.expressionDown.isEmpty
+            }
+            .fullName
+            .l
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala:36]:
```suggestion
            .toSeq
```

File: querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      title = "NOP trust manager used to initialize SSL context",
+      description = "Traffic sent using this context can be intercepted by attackers on the same network",
+      score = 5,
+      withStrRep({ cpg =>
+        val nopTrustManagerFullNames =
+          cpg.typeDecl
+            .isExternal(false)
+            .filter(_.inheritsFromTypeFullName.contains("javax.net.ssl.X509TrustManager"))
+            .filter { node =>
+              node.method.name("checkClientTrusted").block.expressionDown.isEmpty ||
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala:32]:
```suggestion
              node.method.nameExact("checkClientTrusted").block.expressionDown.isEmpty ||
```

File: querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+              node.method.name("checkServerTrusted").block.expressionDown.isEmpty
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/kotlin/NetworkCommunication.scala:33]:
```suggestion
              node.method.nameExact("checkServerTrusted").block.expressionDown.isEmpty
```

**Review by maltek — APPROVED** [review]
this is kind of interesting because we have a Java query for the same thing. If we get more of those we should try to see how much code can be shared between them.

**Comment by ursachec** [comment]
@maltek true, i remember seeing it and then forgot about it. re/ sharing code, it seems like the logical step forward (this started to become more and more clear while writing android queries, most of which require the same manifest/build.gradle parsing logic)

**Pull request merged**